app.R

```diff
@@ -258,26 +257,20 @@ server <- function(input, output, session) {
     return(NULL)
   })
   
-  observe({
-    #if both city and plot type are selected, clear the message
+  output$no_city_message <- renderText({
     if (length(input$locations) > 0 && length(input$ui_parts) > 0) {
-      output$no_city_message <- renderText(NULL)
+      return(NULL)
     }
```

Comment:
As this is the default, valid case, this condition can be omitted and the last default value will be applied then.